pyproject.toml

```diff
@@ -0,0 +1,44 @@
+[project]
+name = "era5grib-parallel"
+authors = [
+  {name = "Chermelle Engel", email="chermelle.engel@anu.edu.au"},
+  {name = "Davide Marchegiani", email="davide.marchegiani@anu.edu.au"},
+]
+maintainers = [
+    { name = "ACCESS-NRI", email = "access.nri@anu.edu.au" }
+]
+description = "Package to set up era5 grib files for use within ACCESS-RAM suites."
+license = { file = "LICENSE" }
+readme = "README.md"
+keywords = ["ACCESS-RAM", "Regional Nesting Suite", "era5grib"]
+dynamic = ["version"]
+dependencies = [
+    "python >=3.10,<=3.12",
```

Comment:
Oh right, yeah I think this should be a specific version. We're using `3.10` in `payu`

https://github.com/ACCESS-NRI/payu-condaenv/blob/main/env.yml#L7

Reply:
Yeah, this is not the right file for that. 
This sets the python dependency for the `era5grib_parallel` `conda` package (similarly to what is done for the [replace_landsurface](https://github.com/ACCESS-NRI/replace_landsurface/blob/ec907101f50f5c045d074726c7c2b7aa35bcad72/pyproject.toml#L17)).

The deployment environment is set in the [access-ram-condaenv env.yaml](https://github.com/ACCESS-NRI/access-ram-condaenv/blob/5ea870b5d5f560bdb12a193fa6bf8d04f476cd96/env.yml#L11), and we are also setting `python=3.10` (similar thing is done for the dev deployed environment).

I would personally update this to `python=3.11`, but I have no strong feelings about it. 

An important point is:
Why do we only specify the `python`'s version? Shouldn's we also specify the versions for the other dependencies there?